src/components/screens/AddAssetScreen/AddAssetScreen.tsx

```diff
@@ -0,0 +1,411 @@
+import { logger } from "config/logger";
+import {
+  MANAGE_ASSETS_ROUTES,
+  ManageAssetsStackParamList,
+} from "config/routes";
+import { PricedBalance, SearchAssetResponse } from "config/types";
+import { useAuthenticationStore } from "ducks/auth";
+import { formatAssetIdentifier } from "helpers/balances";
+import { pxValue } from "helpers/dimensions";
+import { isContractId } from "helpers/soroban";
+import useAppTranslation from "hooks/useAppTranslation";
+import { useBalancesList } from "hooks/useBalancesList";
+import { useClipboard } from "hooks/useClipboard";
+import useColors from "hooks/useColors";
+import useDebounce from "hooks/useDebounce";
+import useGetActiveAccount from "hooks/useGetActiveAccount";
+import { ToastOptions, useToast } from "providers/ToastProvider";
+import React, { useEffect, useRef, useState } from "react";
+import { ScrollView, TouchableOpacity, View } from "react-native";
+import { handleContractLookup } from "services/indexer";
+import {
+  buildChangeTrustTx,
+  signTransaction,
+  submitTx,
+} from "services/stellar";
+import { searchAsset } from "services/stellarExpert";
+
+type AddAssetScreenProps = NativeStackScreenProps<
+  ManageAssetsStackParamList,
+  typeof MANAGE_ASSETS_ROUTES.ADD_ASSET_SCREEN
+>;
+
+enum PageStatus {
+  IDLE = "idle",
+  LOADING = "loading",
+  SUCCESS = "success",
+  ERROR = "error",
+}
+
+const checkIfUserHasTrustline = (
+  currentBalances: (PricedBalance & {
+    id: string;
+  })[],
+  assetCode: string,
+  issuer: string,
+) => {
+  const balance = currentBalances.find((currentBalance) => {
+    const formattedCurrentBalance = formatAssetIdentifier(currentBalance.id);
+
+    return (
+      formattedCurrentBalance.assetCode === assetCode &&
+      formattedCurrentBalance.issuer === issuer
+    );
+  });
+
+  return !!balance;
+};
+
+const formatSearchAssetRecords = (
+  records:
+    | SearchAssetResponse["_embedded"]["records"]
+    | FormattedSearchAssetRecord[],
+  currentBalances: (PricedBalance & {
+    id: string;
+  })[],
+): FormattedSearchAssetRecord[] =>
+  records
+    .map((record) => {
+      if ("assetCode" in record) {
+        return {
+          ...record,
+          hasTrustline: checkIfUserHasTrustline(
+            currentBalances,
+            record.assetCode,
+            record.issuer,
+          ),
+        };
+      }
+
+      const formattedTokenRecord = record.asset.split("-");
+      const assetCode = formattedTokenRecord[0];
+      const issuer = formattedTokenRecord[1] ?? "";
+
+      return {
+        assetCode,
+        domain: record.domain ?? "",
+        hasTrustline: checkIfUserHasTrustline(
+          currentBalances,
+          assetCode,
+          issuer,
+        ),
+        issuer,
+        isNative: record.asset === "XLM",
+      };
+    })
+    .sort((a) => {
+      if (a.hasTrustline) {
+        return -1;
+      }
+
+      return 1;
+    });
+
+const AddAssetScreen: React.FC<AddAssetScreenProps> = ({ navigation }) => {
```

Comment:
@Brunonascdev do you think it would be worth to extract some of the logic from this screen to a hook or store?